modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc

```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-    credentialsName: "worker1-bmc"
-  bootMACAddress: <nic1_mac_address>
-  externallyProvisioned: false <2>
-  customDeploy:
-    method: install_coreos
-  online: true
-  userData:
-    name: worker-user-data-managed
-    namespace: openshift-machine-api
-----
-<1> You can only use bare-metal host drivers that support virtual media networking booting, for example `redfish-virtualmedia` and `idrac-virtualmedia`.
-<2> You must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`. The default value is `false`.
+    address: <protocol>://<bmc_url> <7>
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <8>
+    username: <bmc_username> <9>
+    password: <bmc_password> <9>
+  rootDeviceHints:
+    deviceName: <root_device_hint> <10>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <1>
+----
++
+--
+<1> Replace `<num>` for the machine and worker numbers of the bare metal nodes in the `name`, `credentialsName`, and the `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.15/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[Optional: Configuring host network interfaces] for details on configuring NMState syntax.
+<3> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown:
++
+[source,yaml]
+----
+---
+   interfaces:
+   - name: <nic_name>
+     type: ethernet
+     state: up
+     ipv4:
+       enabled: false
+     ipv6:
+       enabled: false
+----
+<4> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>`, and `<next_hop_nic1_name>` with appropriate values.
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
+<8> To skip certificate validation, set `disableCertificateVerification` to true.
+<9> Replace `<bmc_username>` and `<bmc_password>` with the string of the BMC user name and password.
+<10> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint.
```

Comment:
This field has many variants, the device path is the least recommended one. Do we have any explicit root device hints documentation anywhere to link to?

Reply:
@dtantsur, is this the [documentation](https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#root-device-hints_ipi-install-installation-workflow) you're referring to?

Reply:
Yes, exactly. Maybe link to it?

Reply:
Updated As Suggested.

Reply:
I cant find the link to the root device hint documentation